### ISSUE TITLE
Error in EmailReadImap when email is too long

### ISSUE THREAD

**Issue by Aracon** [n8n-io/n8n#2091]
**Describe the bug**
When email is too long and credentials are correct, EmailReadImap fails with error in console and endless execution in web interface. Log fragment below.

**To Reproduce**
1. Create credentials with long email (my example: notifications-receiver@domain1111111.ru). Credentials name is the same as email. Credentials should be correct (if they are incorrect, node shows login error as expected).
2. Create ReadImapNode with these credentials
3. Run this node

When I use shorter email (send@domain1111111.ru), node works correctly.

**Expected behavior**
Node executes or shows some error in web interface.

**Environment (please complete the following information):**
 - OS: Ubuntu Linux 20.04
 - n8n Version 0.134.0 from Docker container
 - Node.js Version: using n8n from Docker container
 - Database system: default
 - Operation mode: default

**Additional context**
Log fragment (from console with "sudo docker-compose up" running):

```
n8n_1      | (node:60) UnhandledPromiseRejectionWarning: Error: command or literal size is too large
n8n_1      |     at Connection._resTagged (/usr/local/lib/node_modules/n8n/node_modules/imap/lib/Connection.js:1502:11)
n8n_1      |     at Parser.<anonymous> (/usr/local/lib/node_modules/n8n/node_modules/imap/lib/Connection.js:194:10)
n8n_1      |     at Parser.emit (events.js:315:20)
n8n_1      |     at Parser._resTagged (/usr/local/lib/node_modules/n8n/node_modules/imap/lib/Parser.js:175:10)
n8n_1      |     at Parser._parse (/usr/local/lib/node_modules/n8n/node_modules/imap/lib/Parser.js:139:16)
n8n_1      |     at Parser._tryread (/usr/local/lib/node_modules/n8n/node_modules/imap/lib/Parser.js:82:15)
n8n_1      |     at TLSSocket.Parser._cbReadable (/usr/local/lib/node_modules/n8n/node_modules/imap/lib/Parser.js:53:12)
n8n_1      |     at TLSSocket.emit (events.js:315:20)
n8n_1      |     at emitReadable_ (internal/streams/readable.js:569:12)
n8n_1      |     at processTicksAndRejections (internal/process/task_queues.js:79:21)
n8n_1      | (Use `node --trace-warnings ...` to show where the warning was created)
n8n_1      | (node:60) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
n8n_1      | (node:60) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by tennox** [comment]
I want to point out that there's two parts to this issue:
the error itself (email too large)
and the unhandled promise rejection, which I also encountered with a totally different error, and which has the problematic consequence that the workflow keeps running and is not stopped with error.

**Comment by tennox** [comment]
I have submitted a PR to address the second part

**Comment by Joffcom** [comment]
Hey @Aracon & @tennox,

This one should be fixed in the latest release (0.169.0), I am going to mark this one as closed if it happens again let us know and we can always open it back up.
